src/EventSubscriber/DeepLinkSubscriber.php

```diff
@@ -63,7 +63,9 @@ public function onKernelRequest(GetResponseEvent $event): void
         $settings = $this->extractSettingsParameters($request);
         $request->getSession()->set(State::SESSION_SETTINGS_NAME, $settings);
 
-        $event->setResponse($this->responseFactory->createRoutedRedirectResponse('settings'));
+        $event->setResponse($this->responseFactory->createRedirectResponse(
```

Comment:
Previously the redirect worked because `settings` was the actual URL we redirect to.
Now I've chosen to use the framework's URL generator in order to generate the URL for the proper route (which, coincidentally, is also called `settings`) to make everything more robust.